docs: explain text field validation (#3640)
vaadin/docs: articles/components/_input-field-common-features.adoc

////
VALIDATION FEATURES
////

> // tag::constraints-intro[]

Author: I removed this snippet because I considered that it's too small to justify reuse.

.Required
[%collapsible]
====

> Required fields are marked with an indicator next to the label, and become invalid if their value is cleared.

Reviewer: ```suggestion
Required fields are marked with an indicator next to the label, and become invalid if their value is first entered and then cleared.
```

[%collapsible]
====
A separate single-character, regular expression can be used to restrict the characters that can be entered into the field. Characters that don't match the expression are rejected.

> However, programmatically set values are not subject to this restriction, even if they contain disallowed characters. To validate these values, a pattern constraint should be used additionally.

Author: I wanted to add this note but I've realized there is a complication. ComboBox and MultiSelectComboBox both have `setAllowedCharPattern`, but they don't have `setPattern` or if they do, it doesn't work, see https://github.com/vaadin/flow-components/issues/6635. 

I'm now not sure. I wonder whether there is actually a use-case for `setAllowedCharPattern` for MultiSelectComboBox. The `required` constraint seems to be only reasonable for this component.

Reviewer: I guess you could have a CB where you should only be able to enter numbers or something. In any case, both pattern and allowChars are really only meaningful for custom entry. I'd mention it as planned for now and see if we can fix 6635.

> However, programmatically set values are not subject to this restriction, even if they contain disallowed characters. To validate these values, a pattern constraint should be used additionally.

Reviewer: ```suggestion
Note, however, that programmatically set values are not subject to this restriction, even if they contain disallowed characters. To validate these values, a pattern constraint can be used in addition.
```
